lib/engine/game/g_18_uruguay/step/route_rptla.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/route'
+
+module Engine
+  module Game
+    module G18Uruguay
+      module Step
+        class RouteRptla < Engine::Step::Route
+          def setup
+            @goods_shipped = 0
+          end
+
+          def actions(entity)
+            return [] unless entity.corporation == @game.rptla
+
+            %w[run_routes choose].freeze
+          end
+
+          def choosing?(_entity)
+            true
+          end
+
+          def choice_name
+            'Attach goods to ships'
+          end
+
+          def choices
+            choices = {}
+            number_of_goods = [@game.number_of_goods_at_harbor, total_ship_capacity?(current_entity)].min
+            number_of_goods.times do |count|
+              choices[count] = '1 Good' if count.zero?
+              choices[count] = (count + 1).to_s + ' Goods' if count.positive?
+            end
+            choices
+          end
+
+          def ship_capacity(train)
+            (train.name.scan(/\d/)[0].to_f / 2).ceil
+          end
+
+          def total_ship_capacity?(entity)
+            trains = @game.route_trains(entity)
+            total_capacity = 0
+            trains.each do |train|
+              total_capacity += ship_capacity(train)
+            end
+            total_capacity
```

Comment:
The `sum` method takes a block defining what you want to add up, so you can reduce this code nicely:

```suggestion
            trains.sum { |train| ship_capacity(train) }
```